pkg/koordlet/metricsadvisor/devices/rdma/net.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rdma
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/Mellanox/rdmamap"
+	"github.com/jaypipes/ghw"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metricsadvisor/devices/helper"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+)
+
+func GetNetDevice() (metriccache.Devices, error) {
+	pci, err := ghw.PCI()
```

Comment:
Is it testable on Mac?
If not, please place it into the `_linux.go` and add the build tag.